### PR TITLE
add: support for custom API paths

### PR DESCRIPTION
**Pull request by JettChenT**
This allows users to use custom openAI API paths by setting the environment variable(config) `custom_api_url` to their custom API url address.

#51

### CODE REVIEW

File: workflow/src/image_generation.py [hunk 1]
```diff
@@ -18,6 +18,8 @@
 import openai
 
 openai.api_key = os.getenv("api_key")
+if os.getenv("custom_api_url"):
```
Comment (chrislemke) [inline workflow/src/image_generation.py:21]: `custom_api_url` should also be set using the workflows configuration. Please add an additional field, a small documentation and add it to [this](https://github.com/chrislemke/ChatFred#configure-the-workflow-optional-) part of the docs. Thx!!

**Review by chrislemke — CHANGES_REQUESTED** [review]

File: workflow/src/image_generation.py (hunk 1 above)
Comment (JettChenT) [reply workflow/src/image_generation.py:21]: Hi, I've added this in an additional field and updated the docs.

**Comment by JettChenT** [comment]
Hi, I'm wondering if I can get some help regarding the pre commit CI. Running the pre-commit hook locally skips both the black check and the docformatter check.

**Comment by JettChenT** [comment]
> Hey @JettChenT!
> 
> 1. Could you provide an example URL and
> 2. Could you please merge it to develop and not main so I could test it a bit?
> 3. Please add some information about this to the README.md so user understand how this feature works

The example url is `https://closeai.deno.dev/v1`, which added in to the README.md

**Review by chrislemke — APPROVED** [review]

**Comment by chrislemke** [comment]
Looks like some tests/checks are failing. You can use: https://pre-commit.com to solve them or do it manually. 

Thanks!

**Comment by chrislemke** [comment]
@JettChenT!
Sorry for the late response. Let me have a look. I can do it.

**Comment by chrislemke** [comment]
@JettChenT I just merged your stuff. Thanks for contributing!

**Comment by JettChenT** [comment]
Thanks! Sorry I haven't been able to respond the past few days.

**Pull request merged**